test/unit/tarantool_role_test.lua

```diff
@@ -0,0 +1,555 @@
+    },
+    process_expired_tuple_func = {
+        ok = true,
+        options = {process_expired_tuple = always_true_func_name},
+    },
+    process_while_func = {
+        ok = true,
+        options = {process_while = always_true_func_name},
+    },
+    start_key_func = {
+        ok = true,
+        options = {start_key = always_true_func_name},
+    },
+    start_key_table = {
+        ok = true,
+        options = {start_key = {1, 2, 3}},
+    },
+    tuples_per_iteration_number = {
+        ok = true,
+        options = {tuples_per_iteration = 11},
+    },
+    tuples_per_iteration_invalid = {
+        ok = false,
+        err = "roles.expirationd: options.tuples_per_iteration must be a number",
+        options = {tuples_per_iteration = {"table"}},
+    },
+    vinyl_assumed_space_len_factor_number = {
+        ok = true,
+        options = {vinyl_assumed_space_len_factor = 333},
+    },
+    vinyl_assumed_space_len_factor_invalid = {
+        ok = false,
+        err = "roles.expirationd: options.vinyl_assumed_space_len_factor must be a number",
+        options = {vinyl_assumed_space_len_factor = false},
+    },
+    vinyl_assumed_space_len_number = {
+        ok = true,
+        options = {vinyl_assumed_space_len = 1},
+    },
+    vinyl_assumed_space_len_invalid = {
+        ok = false,
+        err = "roles.expirationd: options.vinyl_assumed_space_len must be a number",
+        options = {vinyl_assumed_space_len = "string"},
+    },
+}
+
+for k, case in pairs(options_cases) do
+    g["test_validate_config_option_" .. k] = function(cg)
+        local cfg = create_valid_required({options = case.options})
+        local status, res = pcall(cg.role.validate, cfg)
+        if case.ok then
+            t.assert_equals(status, true)
+            t.assert_equals(res, true)
+        else
+            t.assert_equals(status, false)
+            t.assert_str_contains(res, case.err)
+        end
+    end
+end
+
+function g.test_apply_config_start_tasks(cg)
+    local task_name1 = "apply_config_test1"
+    local task_name2 = "apply_config_test2"
+    pcall(cg.role.apply, {
+        [task_name1] = {
+            space = g.space.id,
+            is_expired = always_true_func_name,
+        },
+        [task_name2] = {
+            space = g.space.id,
+            is_expired = always_true_func_name,
+            options = {}
+        },
+    }, {is_master = false})
+    fiber.sleep(0.5)
+
+    t.assert_not_equals(expirationd.task(task_name1), nil)
+    t.assert_not_equals(expirationd.task(task_name2), nil)
+end
+
+function g.test_apply_config_cfg_metrics_default(cg)
+    t.skip_if(not g.is_metrics_supported,
+              "metrics >= 0.11.0 is not installed")
+    local task_name1 = "apply_config_test1"
+
+    cg.role.apply({
+            [task_name1] = {
+                space = g.space.id,
+                is_expired = always_true_func_name,
+            },
+    }, {is_master = false})
+
+    fiber.sleep(0.5)
```

Comment:
Please, replace the sleep with `helpers.retrying`. It will make test less flaky.